### ISSUE TITLE
No method matching reset(::Cudadrv.CuPrimaryContext)

### ISSUE THREAD

**Issue by una-dinosauria** [JuliaGPU/CUDArt.jl#93]
```julia
julia> using CUDArt

julia> result = devices(dev->true) do devlist
           # Code that does GPU computations
       end
WARNING: destroy(ctx::CuContext) is deprecated, use destroy!(ctx) instead.
Stacktrace:
 [1] depwarn(::String, ::Symbol) at ./deprecated.jl:70
 [2] destroy(::CUDAdrv.CuContext) at ./deprecated.jl:57
 [3] device_reset(::Int64) at /home/julieta/.julia/v0.6/CUDArt/src/device.jl:42
 [4] close(::Array{Int64,1}) at /home/julieta/.julia/v0.6/CUDArt/src/device.jl:186
 [5] devices(::##1#3, ::Array{Int64,1}) at /home/julieta/.julia/v0.6/CUDArt/src/device.jl:84
 [6] devices(::Function, ::Function) at /home/julieta/.julia/v0.6/CUDArt/src/device.jl:74
 [7] eval(::Module, ::Any) at ./boot.jl:235
 [8] eval_user_input(::Any, ::Base.REPL.REPLBackend) at ./REPL.jl:66
 [9] macro expansion at ./REPL.jl:97 [inlined]
 [10] (::Base.REPL.##1#2{Base.REPL.REPLBackend})() at ./event.jl:73
while loading no file, in expression starting on line 0
ERROR: MethodError: no method matching reset(::CUDAdrv.CuPrimaryContext)
Closest candidates are:
  reset(::Base.LibuvStream) at stream.jl:1129
  reset(::T<:IO) where T<:IO at io.jl:622
Stacktrace:
 [1] device_reset(::Int64) at /home/julieta/.julia/v0.6/CUDArt/src/device.jl:44
 [2] close(::Array{Int64,1}) at /home/julieta/.julia/v0.6/CUDArt/src/device.jl:186
 [3] devices(::##1#3, ::Array{Int64,1}) at /home/julieta/.julia/v0.6/CUDArt/src/device.jl:84
 [4] devices(::Function, ::Function) at /home/julieta/.julia/v0.6/CUDArt/src/device.jl:74

julia>
```

Seems like a mismatch between CUDArt and the API of CUDAdrv. I get this both on Julia 0.5.2 and 0.6.

**Comment by una-dinosauria** [comment]
Oh strange. I tried to check for a quick fix but I can't even find a `CUDAdrv.reset` method anymore. Does the package need a major overhaul to catch up with `CUDAdrv.jl`?

**Comment by maleadt** [comment]
Yeah, sorry, I broke it. Working on it. Primary context management is tricky though, see JuliaGPU/CUDAdrv.jl#45 and #95

**Comment by maleadt** [comment]
Fixed: ref #95 and 154c497ed21fbc7ff42f18af45197351afee25d6
